### PR TITLE
Add mdx-driven mli documentation comments

### PR DESCRIPTION
**Pull request by mt-caret**
Closes https://github.com/mt-caret/polars-ocaml/issues/5.

Beautiful docs:

image

I've uploaded the current version of the docs here: https://mt-caret.github.io/polars-ocaml/

Also has the following related changes:
- fix bytecode binding generation in polars-ocaml-macros (more details here: https://github.com/tizoc/ocaml-interop/issues/52)
- set requisite environment variables so that expect tests in both Rust and OCaml are more stable

### CODE REVIEW

**Comment by mt-caret** [comment]
I think odoc only generates documentation for things specified in opam files, which currently is grossly underspecified. I fixed this in https://github.com/mt-caret/polars-ocaml/pull/32 but I'm not super eager to rebase through a million PRs so I'll just wait until the PRs are merged before doing more work here.

**Comment by mt-caret** [comment]
While this is nowhere near complete, I'm opening this to review for merging since:
- there's also a fair amount of functionality in this PR
- I want to nudge folks when adding new things to also add docs, so I'm not stuck with the Sisyphean task of adding docs for a continuously growing list of things!

I'll divy up the work of adding more docs into more PRs.

**Review by somilgo — APPROVED** [review]

**Pull request merged**
